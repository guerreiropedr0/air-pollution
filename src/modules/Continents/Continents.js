import { useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import styles from './Continents.module.css';

const Continents = () => {
  const continents = useSelector((state) => Object.values(state.allReducer.continents));

  const links = [
    {
      path: '/africa',
      text: 'Africa',
      id: 0,
    },
    {
      path: '/asia',
      text: 'Asia',
      id: 1,
    },
    {
      path: '/southamerica',
      text: 'South America',
      id: 2,
    },
    {
      path: '/northamerica',
      text: 'North America',
      id: 3,
    },
    {
      path: '/europe',
      text: 'Europe',
      id: 4,
    },
    {
      path: '/oceania',
      text: 'Oceania',
      id: 5,
    },
  ];

  return (
    <ul className={styles.continents}>
      {links.map(({ path, text, id }) => (
        <li key={text}>
          <NavLink to={path}>{text}</NavLink>
          <small>
            {continents[id] && continents[id].length}
            {' '}
            countries
          </small>
        </li>
      ))}
    </ul>
  );
};

export default Continents;
