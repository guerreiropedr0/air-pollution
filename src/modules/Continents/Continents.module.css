.continents {
  margin-top: 1rem;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  justify-items: center;
  gap: 1px;
}

.continents li {
  width: 100%;
  text-align: center;
  position: relative;
  box-shadow: 0 0 0 1px var(--white-color);
}

.continents li a {
  display: block;
  width: 100%;
  padding: 1.5rem;
  height: 100%;
  background-color: transparent;
  font-size: 0.9rem;
}

.continents li a:hover,
.continents li a:active {
  transform: scale(1.2);
}

.continents li:nth-child(4n + 1) {
  background-color: var(--secondary-color);
}

.continents li:nth-child(4n + 2) {
  background-color: var(--tertiary-color);
}

.continents li:nth-child(4n + 3) {
  background-color: var(--primary-color);
}

.continents li:nth-child(4n + 4) {
  background-color: var(--quaternary-color);
}

.continents li small {
  background-color: transparent;
  pointer-events: none;
  position: absolute;
  bottom: 0;
  left: 50%;
  transform: translateX(-50%);
  opacity: 0.8;
}
