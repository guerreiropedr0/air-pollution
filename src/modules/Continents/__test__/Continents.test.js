import {
  render, screen, waitFor,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Continents from '../Continents';
import store from '../../../redux/ConfigureStore';

const MockedContinents = () => (
  <Provider store={store}>
    <BrowserRouter>
      <Continents />
    </BrowserRouter>
  </Provider>
);

describe('Continents', () => {
  it('renders succesfully', async () => {
    render(<MockedContinents />);
    const africa = await screen.findByText('Africa');
    const asia = await screen.findByText('Asia');
    const southamerica = await screen.findByText('South America');
    const northamerica = await screen.findByText('North America');
    const europe = await screen.findByText('Europe');
    const oceania = await screen.findByText('Oceania');
    expect(africa).toMatchSnapshot();
    expect(asia).toMatchSnapshot();
    expect(southamerica).toMatchSnapshot();
    expect(northamerica).toMatchSnapshot();
    expect(europe).toMatchSnapshot();
    expect(oceania).toMatchSnapshot();
  });

  it('populates countries', async () => {
    render(<MockedContinents />);
    await waitFor(() => screen.getByText('58 countries'), {
      timeout: 4000,
    });
    const africaCountries = await screen.findByText('58 countries');
    expect(africaCountries.innerHTML).toBe('58 countries');
    expect(africaCountries).toMatchSnapshot();
  });
});
