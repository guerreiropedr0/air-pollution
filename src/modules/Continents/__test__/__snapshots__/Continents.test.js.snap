// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Continents populates countries 1`] = `
<small>
  58
   
  countries
</small>
`;

exports[`Continents renders succesfully 1`] = `
<a
  class=""
  href="/africa"
>
  Africa
</a>
`;

exports[`Continents renders succesfully 2`] = `
<a
  class=""
  href="/asia"
>
  Asia
</a>
`;

exports[`Continents renders succesfully 3`] = `
<a
  class=""
  href="/southamerica"
>
  South America
</a>
`;

exports[`Continents renders succesfully 4`] = `
<a
  class=""
  href="/northamerica"
>
  North America
</a>
`;

exports[`Continents renders succesfully 5`] = `
<a
  class=""
  href="/europe"
>
  Europe
</a>
`;

exports[`Continents renders succesfully 6`] = `
<a
  class=""
  href="/oceania"
>
  Oceania
</a>
`;
